### ISSUE TITLE
bug: dlsym(RTLD_DEFAULT, syn_attr2entry): symbol not found

### ISSUE THREAD

**Issue by lucazz** [folke/noice.nvim#608]
### Did you check docs and existing issues?

- [X] I have read all the noice.nvim docs
- [X] I have searched the existing issues of noice.nvim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

v0.9.2

### Operating system/version

macOS 14.0 (23A344)

### Describe the bug

When searching for a string that doesn't exist I'm greeted with the following error message: 

```
 | noice.nvim: ...vim/site/pack/lazy/opt/noice.nvim/lua/noice/util/ffi.lua:38: dlsym(RTLD_DEFAULT, syn_attr2entry): symbol not found
```

From that moment onwards, all other commands I execute generate a slew of the error mentioned above message: 
Here's a clip of it taking place: 
https://github.com/folke/noice.nvim/assets/6126301/3419691b-849a-4fbb-9817-39667cac97a8

### Steps To Reproduce

1. Open neovim/lunarvim
2. Search for a term that doesn't exist in the file

### Expected Behavior

Searches should not render this error message. 

### Repro

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  "folke/noice.nvim",
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here
	{
		"folke/noice.nvim",
		event = "VeryLazy",
		dependencies = {
			"muniftanjim/nui.nvim",
			"rcarriga/nvim-notify",
		},
		config = function()
			local spinners = require("noice.util.spinners")
			spinners.spinners["mine"] = {
				frames = {
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
					" ",
				},
				interval = 80,
			}
			require("noice").setup({
				format = {
					spinner = {
						name = "mine",
						hl = "Constant",
					},
				},
				lsp = {
					progress = {
						enabled = true,
						-- format = "lsp_progress",
						format = {
							{ "{data.progress.percentage} ", hl_group = "Comment" },
							{ "{spinner} ", hl_group = "NoiceLspProgressSpinner" },
							{ "{data.progress.title} ", hl_group = "Comment" },
						},
						format_done = {},
						throttle = 1000 / 30,
						view = "mini",
					},
					override = {
						["vim.lsp.util.convert_input_to_markdown_lines"] = true,
						["vim.lsp.util.stylize_markdown"] = true,
						["cmp.entry.get_documentation"] = true,
					},
				},
				presets = {
					bottom_search = true,
					command_palette = true,
					long_message_to_split = true,
					inc_rename = true,
				},
				views = {
					mini = {
						position = {
							row = -1,
							col = "100%",
						},
					},
					cmdline_popup = {
						border = {
							style = "none",
							padding = { 1, 2 },
						},
						filter_options = {},
						win_options = {
							winhighlight = "NormalFloat:NormalFloat,FloatBorder:FloatBorder",
						},
					},
				},
				routes = {
					{
						filter = {
							event = "msg_show",
							kind = "",
							find = "written",
						},
						opts = { skip = true },
					},
				},
			})
		end,
	}
```

**Comment by folke** [comment]
Something must have changed in the homebrew build related to ffi. That error you're getting is about an ffi function is 100% certain in all Neovim versions > 0.5

**Comment by folke** [comment]
Best to report an issue for homebrew

**Comment by folke** [comment]
Fixed on Nightly:

- [ ] https://github.com/neovim/neovim/issues/25487
- [ ] https://github.com/neovim/neovim/issues/25295

**Comment by folke** [comment]
Seems like the fix will be backported to 0.9 https://github.com/neovim/neovim/pull/25489

Closing...
